### PR TITLE
Add test 'CRUD country'

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | Add test 'CRUD country'
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | TEST_PATH="functional/BO/11_international/02_locations/02_countries/01_CRUDCountry.js" URL_FO=yourShopURL npm run specific-test

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/20831)

### CODE REVIEW

**Review by Progi1984 — DISMISSED** [review]

**Review by PierreRambaud — DISMISSED** [review]

File: tests/UI/pages/FO/myAccount/addresses.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+   * Open create new address form
+   * @param page
+   * @returns {Promise<void>}
+   * @constructor
+   */
+  async openNewAddressForm(page) {
+    await this.waitForSelectorAndClick(page, this.createNewAddressLink);
+  }
+
+  async isCountryExist(page, countryName) {
```
Comment (boubkerbribri) [inline tests/UI/pages/FO/myAccount/addresses.js]: Can you add a description to this function

Comment (boubkerbribri) [reply tests/UI/pages/FO/myAccount/addresses.js]: Also, I think this function should be split to 3 functions : 
`isCountryExist` : to check if the country exist (without choosing it) :

```js
  async isCountryExist(page, countryName) {
    let found = false;
    let options = await page.$$eval(
      `${this.countrySelect} option`,
      all => all.map(
        option => ({
          textContent: option.textContent,
          value: option.value,
        })),
    );
    options = await options.filter(option => countryName === option.textContent);
    return options.length !== 0;
  }
```
And `chooseCountry` : to select a country in the form
```js
  async chooseCountry(page, countryName) {
    await this.selectByVisibleText(page, this.countrySelect, countryName);
  }
```

File: tests/UI/pages/BO/international/locations/countries/index.js [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+  async filterTable(page, filterType, filterBy, value) {
+    switch (filterType) {
+      case 'input':
+        await this.setValue(page, this.filterColumn(filterBy), value.toString());
+        break;
+      case 'select':
+        await this.selectByVisibleText(page, this.filterColumn(filterBy), value ? 'Yes' : 'No');
+        break;
+      default:
+      // Do nothing
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/international/locations/countries/index.js]:
```suggestion
      throw new Error(`Filter ${filterBy} was not found`);
```

File: tests/UI/pages/FO/myAccount/addresses.js (hunk 1 above)
Comment (nesrineabdmouleh) [reply tests/UI/pages/FO/myAccount/addresses.js]: Done thx :+1:

File: tests/UI/pages/BO/international/locations/countries/index.js (hunk 2 above)
Comment (nesrineabdmouleh) [reply tests/UI/pages/BO/international/locations/countries/index.js]: Done thx :+1:

**Review by jolelievre — APPROVED** [review]

**Review by boubkerbribri — APPROVED** [review]

**Pull request merged**